lib/parser.rb

```diff
@@ -1,21 +1,31 @@
 require 'json'
 
 class Parser
-  def initialize(file)
-    @file = file
-    raise "File doesn't exist" unless File.exists?(@file)
+  def initialize(filename)
+    @filename = filename
+    raise "File doesn't exist" unless File.exist?(@filename)
   end
 
   def print_first_line
-    file = File.open(@file)
-    first_line = file.readline
-    file.close
+    filename = File.open(@filename)
+    first_line = filename.readline
+    filename.close
     first_line
```

Comment:
Mesmo comentário que fiz sobre a escolha do nome para variável nesse filename. Além disso, dá pra reduzir bastante a declaração de variáveis. Vale lembrar que cada variável é um endereço a mais que a gente vai consumir da memória, então é sempre bom levar em consideração se é necessário mesmo o uso delas ou não.

```suggestion
    File.open(@filename, &:readline)
```